assembl/graphql/schema.py

```diff
@@ -115,11 +116,14 @@ def resolve_root_idea(self, args, context, info):
         return root_thematic
 
     def resolve_vote_session(self, args, context, info):
-
         discussion_phase_id = args.get('discussion_phase_id')
         discussion_phase = models.DiscussionPhase.get(discussion_phase_id)
-        # TODO: see if we can avoid this next(iter( thing with a one-to-one relationship
-        vote_session = next(iter(discussion_phase.vote_session or []), None)
+        require_discussion_phase_permission = make_instance_permissions_querier(discussion_phase, context)
```

Comment:
you really need to check the permissions like this? I thought that the `SecureObjectType` class took care of it @ayyazdaniaryan @vincentfretin ?

Reply:
I was wondering about that

Reply:
(but maybe I miss something, I read the code quickly)

Reply:
`SecureObjectType` only check the READ permission if you do a node query only.

Reply:
For other queries, we check at least the READ permission on the discussion in the pyramid view.
In the current code afaict, we always use the 'read' permission for the `CrudPermissions.READ` permission type, and permissions-roles mapping is only on discussion, so checking READ permission type on both discussion and on an object is redundant. But you can keep the READ check here, to be safe.